pkgs/tools/system/osquery/default.nix

```diff
@@ -0,0 +1,82 @@
+{ pkgs }:
+
+with pkgs;
+let
+  buildStdenv = overrideCC stdenv llvmPackages.clangUseLLVM;
```

Comment:
Use clangStdenv

Reply:
Unfortunately I've had trouble using `clangStdenv`. I need at least a patch to remove the projects redefinition of `random_shuffle`.

With that patch applied I see the error:
```
/build/source/osquery/core/shutdown.cpp:23:8: error: no type named 'condition_variable' in namespace 'std'
  std::condition_variable request_signal;
  ~~~~~^
```

But I haven't yet determined what I need to do to fix/workaround this.
--- 

```patch
From: Jack Baldry <jack.baldry@grafana.com>
Date: Sat, 21 Jan 2023 19:39:43 +0000
Subject: [PATCH] Remove random_shuffle redefinition

It is already defined in gcc-11.3.0/include/c++/11.3.0/bits/stl_algo.h.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>
---
 libraries/cmake/source/thrift/CMakeLists.txt  |  7 -------
 .../source/thrift/patches/random_shuffle.h    | 21 -------------------
 2 files changed, 28 deletions(-)
 delete mode 100644 libraries/cmake/source/thrift/patches/random_shuffle.h

diff --git a/libraries/cmake/source/thrift/CMakeLists.txt b/libraries/cmake/source/thrift/CMakeLists.txt
index 0b224a159..6f0d824dd 100644
--- a/libraries/cmake/source/thrift/CMakeLists.txt
+++ b/libraries/cmake/source/thrift/CMakeLists.txt
@@ -83,13 +83,6 @@ function(thriftMain)
     "${library_root}/src/thrift/transport/TWebSocketServer.h"
   )
 
-  if(PLATFORM_POSIX)
-    set(tsocket_pool_compile_options "-include;random_shuffle.h")
-
-  elseif(PLATFORM_WINDOWS)
-    set(tsocket_pool_compile_options "/FIrandom_shuffle.h")
-  endif()
-
   set_source_files_properties("${library_root}/src/thrift/transport/TSocketPool.cpp" PROPERTIES
     COMPILE_OPTIONS
       "${tsocket_pool_compile_options}"
diff --git a/libraries/cmake/source/thrift/patches/random_shuffle.h b/libraries/cmake/source/thrift/patches/random_shuffle.h
deleted file mode 100644
index 844ffe3e9..000000000
--- a/libraries/cmake/source/thrift/patches/random_shuffle.h
+++ /dev/null
@@ -1,21 +0,0 @@
-/**
- * Copyright (c) 2014-present, The osquery authors
- *
- * This source code is licensed as defined by the LICENSE file found in the
- * root directory of this source tree.
- *
- * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
- */
-
-#include <algorithm>
-#include <random>
-
-namespace std {
-
-template <class RandomIt>
-void random_shuffle(RandomIt first, RandomIt last) {
-  std::random_device rng;
-  std::mt19937 urng(rng());
-  std::shuffle(first, last, urng);
-}
-} // namespace std
-- 
2.38.1
```

Reply:
Since @jdbaldry tried but didn't succeed to use `clangStdenv`, i propose to go with the current implementation. Also, this is something we could improve in futures PRs (without any breaking changes).